### ISSUE TITLE
BottlerocketOS v1.7.0 not supported by AWS Inspector

### ISSUE THREAD

**Issue by dusansusic** [bottlerocket-os/bottlerocket#2056]
Hi, I've noticed that latest BottlerocketOS images are unsupported by AWS Inspector.
I am curious what broke Inspector support, because v1.6.2 is supported.

**Comment by jpculp** [comment]
Hi @dusansusic, thanks for reaching out! Officially Bottlerocket is not supported by AWS Inspector yet, although it is something we are looking at and is on our roadmap. Can you give us a little more background about how you were using it before? I suspect the issues you are seeing might be stemming from the fact that Bottlerocket hosts now report their platform as Bottlerocket in the SSM agent.

**Comment by dusansusic** [comment]
@jpculp Running multiple EKS clusters with managed node groups with BottlerocketOS AMI. Each node has `arn:aws:iam::aws:policy/AmazonSSMManagedInstanceCore` policy attached. And it just worked with v1.6.2 and all previous versions, AWS Inspector showed findings each time. 
With the same configuration, v1.7.0 doesn't work.

**Comment by jpculp** [comment]
@dusansusic, Amazon Inspector leverages the AWS Systems Manager Agent (SSM Agent) to collect data. Prior to Bottlerocket 1.7.0 hosts would appear as Amazon Linux 2 (the base OS of the control container) in Systems Manager and (if enabled) inventory the packages present on the Control Container. We worked closely with the SSM team and with the release of Bottlerocket 1.7.0 and SSM Agent 3.1.1141.0, Bottlerocket hosts now properly report their platform as Bottlerocket and can provide an inventory of the Bottlerocket host packages rather than the Control Container’s. Amazon Inspector is yet to add support for the Bottlerocket platform (on roadmap for '22).

Although not advised, you could restore the original behavior by simply downgrading your control container to the previous release, v0.5.5, or rebuilding the control container using the latest code and SSM Agent 3.1.821.0. Keep in mind that this configuration would have AWS Inspector basing its findings on the security posture of the Control Container as an Amazon Linux 2 host, and not anything Bottlerocket-specific.

**Comment by bcressey** [comment]
We're working with the AWS Inspector team to add first-class support for Bottlerocket - meaning that its vulnerability scan would be aware of our [security advisories](https://github.com/bottlerocket-os/bottlerocket/security/advisories?state=published) and flag nodes that are not running versions of Bottlerocket that contain the fix.

No ETA yet but it's definitely a priority.

**Comment by dusansusic** [comment]
@bcressey good to hear that support is coming! Thanks!

**Comment by misterek** [comment]
Hi @bcressey !  Any idea when inspector reporting might be coming?

**Comment by bcressey** [comment]
It's tentatively slotted into the 1.10.0 release, roughly August. Please let me know if that doesn't work for you since we can try to account for that in the relative prioritization.

**Comment by misterek** [comment]
Thanks!  Earlier would be helpful, but August should be sufficient. Right now we are checking the current version of bottlerocket vs the version we are running (which is updated with BottleRocketUpdater), which ends up being a completely separate reporting mechanism than Inspector.

Anyway, inspector support will be very welcome, and I'll take it as soon as I can!  Thank you for the update!

**Comment by misterek** [comment]
Ok, last time I'll ask a question until August.   I just wanted to double check, but I assume this will support the new Inspector?  Or will it just be classic?  Or Both?

**Comment by stockholmux** [comment]
@misterek Hey - sorry for the radio silence on this. I know there are folks working on this in the short term.

**Comment by misterek** [comment]
No worries at all!  Last I heard it was slated for 1.10 in August/September.   Thanks!

**Comment by stockholmux** [comment]
@misterek FWIW, it's not exactly aligned to a versioned release but I would expect it on that timeline-ish.

**Comment by misterek** [comment]
Sounds good.  Just need to be able to answer the question "how do you know when there is a security patch available", which, admittedly, is an interesting question in and of itself with bottle rocket.   Right now we just watch the GitHub security advisories and releases, but having it in a tool we already use would be fantastic.

**Comment by misterek** [comment]
@stockholmux Very much apologize for continuing to harp on this, but just wanted to check in and see if Inspector support might be going well.

Also (i can open up a separate issue), how about aws inventory?  I have mixed feelings on this due to the nature of bottlerocket, but I feel like some day, someone will ask.

**Comment by bcressey** [comment]
@misterek it looks like Inspector support will arrive by end of September, though that's gated on internal release processes that can be hard to predict.

You'd asked earlier about new vs. classic Inspector - only the "new" version will be supported.

Re: AWS Inventory - do you mean the inventory feature under AWS Systems Manager, or something else? (It looks like there are a few open source tools with that name.)

For the SSM feature - that's actually supported today and the Inspector integration builds on top of that. In a nutshell, it relies on the SSM agent reporting up host inventory, and the Inspector service then maps that inventory to an `updateinfo.xml` feed. That feed is just the advisory data scraped from this project and cleaned up into a standard format, and it gets refreshed every time there's a new release.

**Comment by misterek** [comment]
@bcressey Perfect. I did mean the SSM feature (and hadn't tried it quite yet on bottlerocket).  Sounds like you folks have covered all the things I was concerned about.  Thanks a ton, and thanks again for your wonderful work on this project.

**Comment by kdaula** [comment]
Bottlerocket is now supported by AWS inspector in commercial regions.

**Comment by dusansusic** [comment]
@kdaula is this supported now for BottlerocketOS 1.9.2 or 1.10.0 which isn't released yet? In Inspector console BottlerocketOS 1.9.2 is still listed as unsupported OS.

**Comment by arnaldo2792** [comment]
@dusansusic this should be supported in versions >= 1.7.0. It takes some time for Inspector to show up the vulnerabilities. I have an instance running a custom build of Bottlerocket, and it shows no findings for it since I built it from the tip of develop:

```bash
[ssm-user@control]$ apiclient get os
{
  "os": {
    "arch": "x86_64",
    "build_id": "27593f8d",
    "pretty_name": "Bottlerocket OS 1.10.0 (aws-k8s-1.21)",
    "variant_id": "aws-k8s-1.21",
    "version_id": "1.10.0"
  }
}
```

You also need to make sure that the control container is running and the SSM agent started correctly, otherwise Inspector won't scan the host.

**Comment by dusansusic** [comment]
Understood. Control container is running and SSM agent is started, all instances have necessary roles. Still, instances are shown as ones with Unsupported OS. Running v1.9.2.

**Comment by arnaldo2792** [comment]
Could you please share with us the AMI id you are using? I tested inspector in a few regions, with different AMIs and versions, and Inspector does detect the hosts. I also tested enabling inspector in a region with existing 1.9.2 hosts and it did detect them.

**Comment by dusansusic** [comment]
@arnaldo2792 it seems it started to scan instances with more than 2 days of delay, ami id: `ami-0e9e6e36c48fb0e8a` (us-east-2 region). Just to be sure, on v1.9.2 there's no single vulnerability shown, is this correct information?

**Comment by bcressey** [comment]
1.9.2 is the most recent release as of today. 1.10.0 should be out soon, and will bring its own set of non-critical security fixes.

**Comment by arnaldo2792** [comment]
>  it seems it started to scan instances with more than 2 days of delay,

Can you check the `Schedule expression` field in your SSM association? For me that's 30 min, you can find it at AWS Systems manager -> State Manager -> <Association ID>

**Comment by dusansusic** [comment]
Disabled/Enabled inspector and it started to work as expected again. Thanks @bcressey @arnaldo2792

**Comment by joebowbeer** [comment]
@bcressey to be clear, only AWS Inspector v2 is supported, right?

**Comment by bcressey** [comment]
@joebowbeer right!
